web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -148,32 +57,40 @@ interface BuyCreditsModalProps extends RegenModalProps {
 }
 
 export interface BuyCreditsValues {
-  retirementBeneficiary: string;
+  retirementNote: string;
   stateProvince: string;
   country: string;
   postalCode: string;
   retirementAction: string;
   creditCount: number;
+  price: number;
+  askDenom: string;
+  batchDenom: string;
+  sellOrderId: string;
 }
 
 const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
   open,
+  onSubmit,
   onClose,
   onTxQueued,
   initialValues,
   project,
   apiServerUrl,
   imageStorageBaseUrl,
 }) => {
-  const styles = useStyles();
+  const styles = useBuyCreditsModalStyles();
+  const theme = useTheme();
   const walletContext = useWallet();
 
   const submit = async (values: BuyCreditsValues): Promise<void> => {
     const recipient = 'regen18hj7m3skrsrr8lfvwqh66r7zruzdvp6ylwxrx4'; // test account
     const amount = values.creditCount;
-    if (walletContext.signSend && walletContext.broadcast) {
+    if (onSubmit) {
+      onSubmit(values);
+    } else if (walletContext.signSend && walletContext.broadcast) {
```

Comment:
Temporary call submits so visualize the whole flow without doing an actual transaction.